tabrepo/repository/ensemble_mixin.py

```diff
@@ -1,83 +1,286 @@
     def evaluate_ensemble(
         self,
-        datasets: list[str],
+        dataset: str,
+        fold: int,
         configs: list[str] = None,
         *,
+        time_limit: float = None,
         ensemble_cls: Type[EnsembleScorer] = EnsembleScorerMaxModels,
         ensemble_kwargs: dict = None,
         ensemble_size: int = 100,
         rank: bool = True,
-        folds: list[int] | None = None,
-        backend: str = "ray",
-    ) -> Tuple[pd.Series, pd.DataFrame]:
+        fit_order: Literal["original", "random"] = "original",
+        seed: int = 0,
+    ) -> Tuple[pd.DataFrame, pd.DataFrame]:
         """
-        :param datasets: list of datasets to compute errors on.
-        :param configs: list of config to consider for ensembling. Uses all configs if None.
-        :param ensemble_size: number of members to select with Caruana.
-        :param ensemble_cls: class used for the ensemble model.
-        :param ensemble_kwargs: kwargs to pass to the init of the ensemble class.
-        :param rank: whether to return ranks or raw scores (e.g. RMSE). Ranks are computed over all base models and
-        automl framework.
-        :param folds: list of folds that need to be evaluated, use all folds if not provided.
-        :param backend: Options include ["native", "ray"].
-        :return: Tuple:
-            Pandas Series of ensemble test errors per task, with multi-index (dataset, fold).
-            Pandas DataFrame of ensemble weights per task, with multi-index (dataset, fold). Columns are the names of each config.
+        Evaluates an ensemble of a list of configs on a given task (dataset, fold).
+
+        Parameters
+        ----------
+        dataset: str
+            The dataset to evaluate
+        fold: int
+            The fold of the dataset to evaluate
+        configs: list[str], default = None
+            The list of configs to consider for ensembling.
+            If None, will use all configs.
+            Models will be simulated as being fit in the order specified in `fit_order`.
+        time_limit: float, default = None
+            The time limit of the ensemble.
+            Will only consider the first N models in `configs` whose cumulative time limit is less than `time_limit`.
+        ensemble_cls: Type[EnsembleScorer], default = EnsembleScorerMaxModels
+            The ensemble method to use.
+        ensemble_kwargs: dict, default = None
+            The ensemble method kwargs.
+        ensemble_size: int, default = 100
+            The number of ensemble iterations.
+        rank: bool, default = True
+            If True, additionally calculates the rank of the ensemble result.
+        fit_order: Literal["original", "random"], default = "original"
+            Whether to simulate the models being fit in their original order sequentially or randomly.
+        seed: int, default = 0
+            The random seed used to shuffle `configs` if `fit_order="random"`.
```

Comment:
I am not sure whether we should have this here, wouldnt it be better if the caller the configuration randomly if he/she wants?

Reply:
I think either works, but I'm preferring keeping it for the sake that it isn't much code to add and is a nice piece of logic since it avoids the user accidentally implementing it incorrectly. Plus, with configs=None by default, they don't necessarily know what configs are available on a given task, so being able to shuffle them automatically is nice. The default is `original`, so it doesn't alter the default user experience.